src/main/java/com/dataloom/datastore/services/LinkingService.java

```diff
@@ -80,7 +83,7 @@ public UUID link( UUID linkedEntitySetId, Set<Map<UUID, UUID>> linkingProperties
 
         // Warning: We assume that the restrictions on links are enforced/validated as specified in LinkingApi. In
         // particular, from now on we work on the assumption that only identical property types are linked on.
-        initializeComponents( linkIndexedByEntitySets.keySet(), linkIndexedByPropertyTypes, linkIndexedByEntitySets );
+        initializeComponents( dsm.getLatestSyncId( linkIndexedByEntitySets.keySet() ), linkIndexedByPropertyTypes, linkIndexedByEntitySets );
```

Comment:
this is unnecessary--the search across indices function in SearchService takes the same input, and uses the current sync id for each of the entity sets. So it will still expect Set<UUID> of entity set ids.